server/src/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -854,15 +856,18 @@ public SnapshotPolicyVO createPolicy(CreateSnapshotPolicyCmd cmd, Account policy
                 throw new InvalidParameterValueException("Max number of snapshots shouldn't exceed the " + message + " level snapshot limit");
             }
         }
-        SnapshotPolicyVO policy = _snapshotPolicyDao.findOneByVolumeInterval(volumeId, intvType);
-        if (policy == null) {
-            policy = new SnapshotPolicyVO(volumeId, cmd.getSchedule(), timezoneId, intvType, cmd.getMaxSnaps(), display);
-            policy = _snapshotPolicyDao.persist(policy);
-            _snapSchedMgr.scheduleNextSnapshotJob(policy);
-        } else {
-            try {
+
+        final GlobalLock createSnapshotPolicyLock = GlobalLock.getInternLock("createSnapshotPolicy_" + volumeId);
+        createSnapshotPolicyLock.lock(5);
```

Comment:
The return value is not checked (`boolean`) which should handle the case when the lock could not be acquired.

Reply:
@marcaurele : Do we just need to add the debug logger if lock is not acquired ?
Please confirm.

Reply:
If the lock is not acquired you have to stop processing the request and return a failure to the user. Look at other part of the code using that lock mechanism.

Reply:
@marcaurele : Incorporated the changes suggested by you for handling the condition of lock not acquired. In case if lock is not acquired, null value will be returned.
In CreateSnapshotPolicyCmd class, inside execute method from where this method is called, we have already handled the null condition, and in case of null value exception will be thrown.

Please review the latest code.